twig/compiler.py

```diff
@@ -0,0 +1,14 @@
+from twig.utils.compiler import generate_contract_types, generate_inline_sources
+
+
+class Compiler:
+    def __init__(self, sources, backend):
+        self.sources = sources
+        self.backend = backend
+        self.output = self.backend.compile(self.sources)
```

Comment:
I think this should be a computed property rather than populating it eagerly during initialization.  No strong reason other than compilation often being expensive.